tests/test_script_helper.py

```diff
@@ -1,222 +1,284 @@
-        os.chdir(cwd)
-        instance_created = h.original_cell if purpose == "reduction" else h.control_cell
+        import_3d.instantiate(instance_created)
+        # this function should be included in the model.hoc file given as a parameter
+        instance_created.complete_full_model_creation()
+    elif create_type in ('bbp', 'bbpactive'):
+        h('{instance_name} = new {model_obj_name}(1, "{morphology_file}")'.format(
+            morphology_file=morphology_file, **model))
+        h('{instance_name} = {instance_name}.CellRef'.format(**model))
 
-    elif create_type in ['hay', 'almog','allen']:
-        h(instance_name + " = new " + model_obj_name + '("' + morphology_file +'")')
-        instance_created = h.original_cell if purpose == "reduction" else h.control_cell
+    elif create_type in ('bbpnew', ):
+        with cd(morphology_file[:morphology_file.rindex('/')]):
+            h('{instance_name} = new {model_obj_name}(0)'.format(**model))
+    elif create_type in ('hay', 'almog', 'allen'):
+        h('{instance_name} = new {model_obj_name}("{morphology_file}")'.format(
+            morphology_file=morphology_file, **model))
 
+    return getattr(h, instance_name)
 
-    return instance_created 
 
+SynapseLocation = namedtuple('SynapseLocation', 'type, section_num, x')
 
 
+def create_synapses(cell_instance,
+                    synapse_file,
+                    seed_for_random_synapse_classification=SEED_FOR_RANDOM_SYNAPSE_CLASSIFICATION,
+                    percentage_of_excitatory_synapses=PERCENTAGE_OF_EXCITATORY_SYNAPSES):
+    '''Creates the synapses according to the synapse locations file
 
-def create_synapses(cell_instance):
-    ''' 
-    Creates the synapses according to the synapse locations file given to the program and attaces them to the given cell instance. 
-    Returns the new synapses_list, netstims_list, netcons_list, and randoms_list (the activity generators for the NetStims).
-    Supports two types of synapses, to support more types CODE CHANGES are required (lines 156-180).
+    attaches them to the given cell instance.
+    Returns the new synapses_list, netstims_list, netcons_list, and
+    randoms_list (the activity generators for the NetStims).
+
+    TODO: Supports two types of synapses, to support more types CODE CHANGES are required
     '''
-    f = open(SYNAPSE_FILE)
-    synapses_list = []
-    netstims_list = []
-    netcons_list = []
-    randoms_list = []
-    
-    random_classifier = random.Random(SEED_FOR_RANDOM_SYNAPSE_CLASSIFICATION)  # the seed used to randomly classify synapses from the file into different types
-    
-    num_of_excitatory_syns = 0
-    num_of_inhibitory_syns = 0
-    
-    # iterates over synapse locations file and creates synapses, NetStims and NetCons
-    orig_synapse_index = 0
-    for line in iter(f):
-        line = line.strip()
-        if line:  # for non-empty lines
+    synapses = []
+    with open(synapse_file) as fd:
```

Comment:
changed this to two passes: one is loading the file, the second is actually adding the synapses - should be easier to debug and deal with file format changes, I think